Objects/typeobject.c

```diff
@@ -6772,6 +6759,31 @@ type_ready_set_new(PyTypeObject *type)
     return 0;
 }
 
+static int
+type_ready_managed_dict(PyTypeObject *type)
+{
+    if (!(type->tp_flags & Py_TPFLAGS_MANAGED_DICT)) {
```

Comment:
For me it would make more sense to only call the function if Py_TPFLAGS_MANAGED_DICT is set, rather than calling the function to do nothing if it's not set.

Reply:
See other comment, lets keep the code consistent and to the skip checks in the helper functions.